core/lib/zksync_core/src/proof_data_handler/request_processor.rs

```diff
@@ -140,7 +143,60 @@ impl RequestProcessor {
                     .await
                     .map_err(RequestProcessorError::ObjectStore)?;
 
+                let system_logs_hash_from_prover =
+                    H256::from_slice(&proof.aggregation_result_coords[0]);
+                let state_diff_hash_from_prover =
+                    H256::from_slice(&proof.aggregation_result_coords[1]);
+                let bootloader_heap_initial_content_from_prover =
+                    H256::from_slice(&proof.aggregation_result_coords[2]);
+                let events_queue_state_from_prover =
+                    H256::from_slice(&proof.aggregation_result_coords[3]);
+
                 let mut storage = self.pool.access_storage().await.unwrap();
+
+                // let header = storage
+                //     .blocks_dal()
+                //     .get_l1_batch_header(l1_batch_number)
+                //     .await
+                //     .unwrap()
+                //     .expect("Proved block without a header");
+
+                let l1_batch = storage
+                    .blocks_dal()
+                    .get_l1_batch_metadata(l1_batch_number)
+                    .await
+                    .unwrap()
+                    .expect("Proved block without metadata");
```

Comment:
this is bad - if the gateway submits a wrong proof (for example for the non existing l1 batch), rather than returning the error, we'll have state keeper crashing, right?

Reply:
@mm-zk the `gateway` is run as a separate component not along with statekeeper, and yes if we get an proof for  non-existing batch it will result in crashing of `gateway` for that batch.